### PR TITLE
issue #2859834 implementation of links in checkout progress block

### PR DESCRIPTION
**Pull request by finnef**

### CODE REVIEW

**Comment by finnef** [comment]
DO: https://www.drupal.org/node/2859834

**Comment by bojanz** [comment]
We also need a setting for enabling this, like the D7 module had.

File: modules/checkout/src/Plugin/Block/CheckoutProgressBlock.php [hunk 1]
```diff
@@ -103,9 +104,20 @@ public function build() {
         $position = 'next';
       }
 
+      // Create breadcrumb style links for active checkout steps.
+      if ($index <= $current_step_index) {
+        $label = Link::createFromRoute($step_definition['label'], 'commerce_checkout.form', [
+          'commerce_order' => $order->id(),
+          'step' => $step_id,
+        ])->toString();
```
Comment (onatalushko) [inline modules/checkout/src/Plugin/Block/CheckoutProgressBlock.php]: maybe ->to Renderable() will be better here ?

Comment (mglaman) [reply modules/checkout/src/Plugin/Block/CheckoutProgressBlock.php]: Should always be renderable, unless that does fail

File: modules/checkout/templates/commerce-checkout-progress.html.twig [hunk 2]
```diff
@@ -14,6 +14,6 @@
 #}
 <ol class="checkout-progress clearfix">
   {% for step in steps %}
-      <li class="checkout-progress--step checkout-progress--step__{{ step.position }}">{{ step.label }}</li>
+      <li class="checkout-progress--step checkout-progress--step__{{ step.position }}">{{ step.label | raw }}</li>
```
Comment (mglaman) [inline modules/checkout/templates/commerce-checkout-progress.html.twig]: If we remove `raw` toRenderable should be fine

**Review by mglaman — CHANGES_REQUESTED** [review]

**Comment by finnef** [comment]
changed to Renderable() and removed | raw.
Also added config checkbox

**Review by mglaman — CHANGES_REQUESTED** [review]
```
    1)
    Drupal\Tests\commerce_checkout\Kernel\ChainCheckoutFlowResolverTest::testCheckoutFlowResolution
    Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for
    commerce_checkout.commerce_checkout_flow.default with the following errors:
    commerce_checkout.commerce_checkout_flow.default:configuration.display_checkout_progress_breadcrumb_links
    missing schema
```

the new option needs to be added to the module's schema.

can we also update a test where it clicks a link and goes back and continues?

**Comment by finnef** [comment]
Added tests and schema.

File: modules/checkout/tests/src/Functional/CheckoutOrderTest.php [hunk 3]
```diff
@@ -155,13 +166,14 @@ public function testRegisterOrderCheckout() {
       'login[register][password][pass2]' => 'pass',
     ], 'Create account and continue');
     $this->assertSession()->pageTextContains('Billing information');
+    // Check breadcrumbs are not links. (the default setting)
+    $this->assertSession()->elementNotExists('css', '.block-commerce-checkout-progress li.checkout-progress--step > a');
```
Comment (mglaman) [inline modules/checkout/tests/src/Functional/CheckoutOrderTest.php:170]: So we assert it's not a link, however we do not prove the feature works in any test

**Review by mglaman — CHANGES_REQUESTED** [review]
Almost thumbs up here. However our tests do not prove this works or document the interaction.

File: modules/checkout/tests/src/Functional/CheckoutOrderTest.php (hunk 3 above)
Comment (finnef) [reply modules/checkout/tests/src/Functional/CheckoutOrderTest.php:170]: The test to use the link to move back and fwd through the checkout steps is in testGuestOrderCheckout. The test for no link (by default) is in testRegisterOrderCheckout.

**Review by mglaman — APPROVED** [review]

File: modules/checkout/tests/src/Functional/CheckoutOrderTest.php (hunk 3 above)
Comment (mglaman) [reply modules/checkout/tests/src/Functional/CheckoutOrderTest.php:170]: Oh duh, wow i totally missed that. All good then!

**Comment by bojanz** [comment]
Continued (rebased) in #732.

**Pull request closed without merging**